File: models/Gpt-3.5-turbo/cpp/code/problems/introductory_problems/Repetitions.cpp

#include <iostream>
#include <string>

using namespace std;

int main() {
    string dna;
    cin >> dna;

    int n = dna.length();
    int longestRepetition = 1;
    int currentRepetition = 1;

    for (int i = 1; i < n; i++) {
        if (dna[i] == dna[i - 1]) {
            currentRepetition++;
        } else {
            longestRepetition = max(longestRepetition, currentRepetition);
            currentRepetition = 1;
        }
    }

    longestRepetition = max(longestRepetition, currentRepetition);
    cout << longestRepetition << endl;

    return 0;
}

File: models/Gpt-3.5-turbo/py/code/introductory_problems/Repetitions.py

# Read the input DNA sequence
sequence = input()

# Initialize variables
max_repetition = 0
current_repetition = 0
prev_char = None

# Iterate through the sequence
for char in sequence:
    # If the current character is the same as the previous one
    if char == prev_char:
        current_repetition += 1
    # If the current character is different
    else:
        # Update the maximum repetition length if necessary
        max_repetition = max(max_repetition, current_repetition)
        # Reset the current repetition length
        current_repetition = 1
        # Update the previous character
        prev_char = char

# Update the maximum repetition length after the last character
max_repetition = max(max_repetition, current_repetition)

# Print the result
print(max_repetition)
